app/main.py

```diff
@@ -2,24 +2,67 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int | float,
+                 max_amount: int | float) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> int | str:
```

Comment:
The `__get__` method in the `IntegerRange` class should return the correct type. Currently, it returns `int | str`, but it should return `int | float` to match the expected types of the descriptor.